packages/opencensus-core/src/trace/model/span-base.ts

```diff
@@ -133,7 +137,17 @@ export abstract class SpanBase implements types.Span {
    * @param value The result of an operation.
    */
   addAttribute(key: string, value: string|number|boolean) {
-    this.attributes[key] = value;
+    this.totalRecordedAttributes++;
+    if (Object.keys(this.attributes.attributeMap).length >=
+        this.activeTraceParams.numberOfAttributesPerSpan) {
+      const attributeKeyToDelete =
+          Object.keys(this.attributes.attributeMap).shift();
+      delete this.attributes.attributeMap[attributeKeyToDelete];
```

Comment:
Why delete an existing attribute rather than just dropping the new attribute? Is that the specified way to do it?

Reply:
This implementation is based on the spec here: https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/TraceConfig.md#limits

> When limits are exceeded, implementations should by default preserve the most recently added values and drop the oldest values.

Reply:
Thanks, makes sense.

Reply:
does shift() guarantee that it returns the oldest?

Reply:
As per my knowledge, integer properties are sorted, others (non-integer) are listed in the creation order. ```shift``` -> Removes the first element from an array. @draffensperger correct me, if I am wrong.

Reply:
I checked on the order by running the following JS in a `node` console:
```
x = {}
x.b = 2
x.a = 1
Object.keys(x)  // gives ['b', 'a'] as expected

y = {}
y.a = 1
y.b = 2
Object.keys(y)  // gives ['a', 'b'] as expected
```

That said, what would you think about adding a unit test to validate this "removes the last attribute" behavior?

Reply:
OK, sweet!